lib/full-site-editing/class-wp-rest-templates-controller.php

```diff
@@ -0,0 +1,426 @@
+			$changes['post_title'] = $request['title'];
+		} elseif ( null !== $template && ! $template->is_custom ) {
+			$changes['post_title'] = $template->title;
+		}
+		if ( isset( $request['description'] ) ) {
+			$changes['post_excerpt'] = $request['description'];
+		} elseif ( null !== $template && ! $template->is_custom ) {
+			$changes['post_excerpt'] = $template->description;
+		}
+
+		return $changes;
+	}
+
+	/**
+	 * Prepare a single template output for response
+	 *
+	 * @param WP_Block_Template $template    Temlate instance.
+	 * @param WP_REST_Request   $request Request object.
+	 *
+	 * @return WP_REST_Response $data
+	 */
+	public function prepare_item_for_response( $template, $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		$result = array(
+			'id'          => $template->id,
+			'theme'       => $template->theme,
+			'content'     => array( 'raw' => $template->content ),
+			'slug'        => $template->slug,
+			'is_custom'   => $template->is_custom,
+			'type'        => $template->type,
+			'description' => $template->description,
+			'title'       => array(
+				'raw'      => $template->title,
+				'rendered' => $template->title,
+			),
+			'status'      => $template->status,
+			'wp_id'       => $template->wp_id,
+		);
+
+		$response = rest_ensure_response( $result );
+		$links    = $this->prepare_links( $template->id );
+		$response->add_links( $links );
+		if ( ! empty( $links['self']['href'] ) ) {
+			$actions = $this->get_available_actions();
+			$self    = $links['self']['href'];
+			foreach ( $actions as $rel ) {
+				$response->add_link( $rel, $self );
+			}
+		}
+
+		return $response;
+	}
+
+
+	/**
+	 * Prepares links for the request.
+	 *
+	 * @since 4.7.0
+	 *
+	 * @param integer $id ID.
+	 * @return array Links for the given post.
+	 */
+	protected function prepare_links( $id ) {
+		$base = sprintf( '%s/%s', $this->namespace, $this->rest_base );
+
+		$links = array(
+			'self'       => array(
+				'href' => rest_url( trailingslashit( $base ) . $id ),
+			),
+			'collection' => array(
+				'href' => rest_url( $base ),
+			),
+			'about'      => array(
+				'href' => rest_url( 'wp/v2/types/' . $this->post_type ),
+			),
+		);
+
+		return $links;
+	}
+
+	/**
+	 * Get the link relations available for the post and current user.
+	 *
+	 * @return array List of link relations.
+	 */
+	protected function get_available_actions() {
+		$rels = array();
+
+		$post_type = get_post_type_object( $this->post_type );
+
+		if ( current_user_can( $post_type->cap->publish_posts ) ) {
+			$rels[] = 'https://api.w.org/action-publish';
+		}
+
+		if ( current_user_can( 'unfiltered_html' ) ) {
+			$rels[] = 'https://api.w.org/action-unfiltered-html';
+		}
+
+		return $rels;
+	}
+
+	/**
+	 * Retrieves the query params for the posts collection.
+	 *
+	 * @return array Collection parameters.
+	 */
+	public function get_collection_params() {
+		$query_params = parent::get_collection_params();
```

Comment:
We should unset the `search`, `page`, and `per_page` parameters if they aren't supported.